### DISCUSSION TITLE
If you expose GUI basecolors and fonts specification to userspace...[customization?]

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1083] [Ideas]
![Example GUI changes](https://i.imgur.com/C9g04os.png)

Perhaps I just have a pet dislike for drab grey GUIs ~ obviously I've been reading main_window.cpp

```
        //-------------------------------------------------
        // Define base colors
        //-------------------------------------------------
        gui_baseCol = gcn::Color(170, 190, 225);
        colSelectorInactive = gcn::Color(200, 170, 230);
        colSelectorActive = gcn::Color(183, 186, 100);
        colTextboxBackground = gcn::Color(220, 220, 220);
```

The font is "mO’sOul .ttf" from the archive at https://www.retro32.com/amiga-resources/220520201028-multi-platform-amiga-fonts-v1-02-download-ascii-truetype-ttf   (one can just rename the file and copy to data/AmigaTopaz.ttf as a working kludge).

I think it'd be a nice 'feature add' to expose these things to userspace (perhaps via amiberry.conf), to allow some customization to the amiberry GUI at runtime (to better visually suit their particular setup/taste). I do say this knowing that font color may likely also need to be added, if users choose dark GUI colors, but that aside it'd make for simplistic changes to GUI look =)

[btw, line 659 - "// Quit entire program via SQL-Quit"  ...made me chuckle 8^]

**Comment by midwan** [comment]
Themes are a good idea, and I was thinking about it as well.
But it will require some work, as the colors and font name/size need to be exposed. Not too hard I believe, at least for a first version.

**Comment by giantclambake** [comment]
Cool bananas, I look forward to testing it =)

**Reply by midwan** [reply]
This is now implemented

**Reply by giantclambake** [reply]
Thanks ;)  See my comment on the commit ~ something I found while testing.

edit: probably just me, seems ok now

**Comment by giantclambake** [comment]
Closing resolved ; as this is now added to master
